cfme/v2v/conversion_hosts.py

```diff
@@ -0,0 +1,240 @@
+        transformation_method = BootstrapSelect("transformationMethod")
+        vddk_library_path = Input(id='vddk-library-path')
+        vmware_ssh_key = TextInput(locator='.//textarea[@id="vmware-ssh-key-input"]')
+        osp_cert_switch = V2VBootstrapSwitch(label='Verify TLS Certificates for OpenStack')
+        osp_ca_cert = TextInput(locator='.//textarea[@id="openstack-ca-certs-input"]')
+
+        @property
+        def is_displayed(self):
+            return self.conv_host_key.is_displayed
+
+        def after_fill(self, was_change):
+            self.parent.configure_btn.click()
+
+    @View.nested
+    class results(View):  # noqa
+        msg = Text('.//h3[contains(@class,"blank-slate-pf-main-action")]')
+
+        @property
+        def is_displayed(self):
+            return self.msg.is_displayed
+
+
+@attr.s
+class ConversionHost(BaseEntity):
+    """Class representing conversion Host
+        Args:
+            target_provider: RHV or OSP
+            cluster: Default
+            hostname: host to be configured
+            conv_host_key: /etc/pki/ovirt-engine/keys/engine_id_rsa on RHV-M.
+            transformation_method: vddk or ssh
+            vddk_library_path: url for vddk
+            vmware_ssh_key: /var/lib/vdsm/.ssh/id_rsa on rhv host
+            osp_ca_cert : CA cert
+
+    """
+    target_provider = attr.ib()
+    cluster = attr.ib()
+    hostname = attr.ib()
+    conv_host_key = attr.ib()
+    transformation_method = attr.ib(validator=attr.validators.in_(["VDDK", "SSH"]))
+    vddk_library_path = attr.ib(default=None)
+    vmware_ssh_key = attr.ib(default=None)
+    osp_cert_switch = attr.ib(default=None)
+    osp_ca_cert = attr.ib(default=None)
+
+    def fill_dict(self):
+        """Generate a dictionary for filling the Conversion host view
+        Returns: dict
+        """
+        provider_type = ("Red Hat Virtualization" if self.target_provider.one_of(RHEVMProvider)
+                         else "Red Hat OpenStack Platform")
+        return {
+            "location": {"provider_type": provider_type, "provider_name": self.target_provider.name,
+                         "cluster": self.cluster},
+            "hosts": {"hostname": self.hostname},
+            "authentication": {"conv_host_key": self.conv_host_key,
+                               "transformation_method": self.transformation_method,
+                               "vddk_library_path": self.vddk_library_path,
+                               "vmware_ssh_key": self.vmware_ssh_key,
+                               "osp_cert_switch": self.osp_cert_switch,
+                               "osp_ca_cert": self.osp_ca_cert}
+        }
+
+    @property
+    def is_host_configured(self):
+        """Wait for conversion host to configure and return success/failure state"""
+        view = navigate_to(self.parent, "All")
+        try:
+            wait_for(
+                lambda: not view.conversion_host_progress.in_progress(self.hostname),
```

Comment:
Suggestion/optional - You can avoid using a `lambda` here, as you're actually calling a function, not a property.
`wait_for(view.conversion_host_progress.in_progress, func_args=[self.hostname], fail_condition=True)`